wasm/jit/jit_arm64.go

```diff
@@ -476,20 +512,141 @@ func (c *arm64Compiler) compileLabel(o *wazeroir.OperationLabel) (skipThisLabel
-	return fmt.Errorf("TODO: unsupported on arm64")
+	x := c.locationStack.peek()
+	y := c.locationStack.stack[int(c.locationStack.sp)-1-o.Depth]
+
+	if err := c.maybeCompileEnsureOnGeneralPurposeRegister(x); err != nil {
+		return err
+	}
+	if err := c.maybeCompileEnsureOnGeneralPurposeRegister(y); err != nil {
+		return err
+	}
+
+	x.register, y.register = y.register, x.register
+	return nil
 }
 
+// compileGlobalGet implements compiler.compileGlobalGet for the arm64 architecture.
 func (c *arm64Compiler) compileGlobalGet(o *wazeroir.OperationGlobalGet) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	c.maybeCompileMoveTopConditionalToFreeGeneralPurposeRegister()
+
+	// Get the address of globals[index] into intReg.
+	intReg, err := c.compileReadGlobalAddress(o.Index)
+	if err != nil {
+		return err
+	}
+
+	var intMov, floatMov obj.As = obj.ANOP, obj.ANOP
+	switch c.f.ModuleInstance.Globals[o.Index].Type.ValType {
+	case wasm.ValueTypeI32:
+		intMov = arm64.AMOVW
+	case wasm.ValueTypeI64:
+		intMov = arm64.AMOVD
+	case wasm.ValueTypeF32:
+		intMov = arm64.AMOVW
+		floatMov = arm64.AFMOVS
+	case wasm.ValueTypeF64:
+		intMov = arm64.AMOVW
+		floatMov = arm64.AFMOVD
+	}
+
+	// "intReg = [intReg + globalInstanceValueOffset] (== globals[index].Val)"
+	c.compileMemoryToRegisterInstruction(
+		intMov,
+		intReg, globalInstanceValueOffset,
+		intReg,
+	)
+
+	// If the value type is float32 or float64, we have to move the value
+	// further into the float register.
+	resultReg := intReg
+	if floatMov != obj.ANOP {
+		resultReg, err = c.allocateRegister(generalPurposeRegisterTypeFloat)
+		if err != nil {
+			return err
+		}
+		c.compileRegisterToRegisterInstruction(floatMov, intReg, resultReg)
+	}
+
+	c.locationStack.pushValueLocationOnRegister(resultReg)
+	return nil
 }
 
+// compileGlobalSet implements compiler.compileGlobalSet for the arm64 architecture.
 func (c *arm64Compiler) compileGlobalSet(o *wazeroir.OperationGlobalSet) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	val := c.locationStack.pop()
+	if err := c.maybeCompileEnsureOnGeneralPurposeRegister(val); err != nil {
+		return err
+	}
+
+	globalInstanceAddressRegister, err := c.compileReadGlobalAddress(o.Index)
+	if err != nil {
+		return err
+	}
+
+	var mov obj.As
+	switch c.f.ModuleInstance.Globals[o.Index].Type.ValType {
+	case wasm.ValueTypeI32:
+		mov = arm64.AMOVW
+	case wasm.ValueTypeI64:
+		mov = arm64.AMOVD
+	case wasm.ValueTypeF32:
+		mov = arm64.AFMOVS
+	case wasm.ValueTypeF64:
+		mov = arm64.AFMOVD
+	}
+
+	// At this point "globalInstanceAddressRegister = globals[index]".
+	// Therefore, this means "globals[index].Val = val.register"
+	c.compileRegisterToMemoryInstruction(
+		mov,
+		val.register,
+		globalInstanceAddressRegister, globalInstanceValueOffset,
+	)
+
+	c.markRegisterUnused(val.register)
+	return nil
+}
+
+// compileReadGlobalAddress adds instruction to read the absolute address of the global instance whose index equals globalIndex,
```

Comment:
try to tighten if it still means the same
"adds instructions to store the absolute address of the global instance at globalIndex into a register"